app/gui/view/src/documentation.rs

```diff
@@ -328,6 +357,22 @@ impl View {
 
             app.frp.show_system_cursor <+ overlay.events.mouse_over;
             app.frp.hide_system_cursor <+ overlay.events.mouse_out;
+
+
+            // === Hover ===
+
+            eval_ model.overlay.events.mouse_over(model.on_hover_start());
+            eval_ model.overlay.events.mouse_out(model.on_hover_end());
+            is_hovered <- any(...);
+            is_hovered <+ model.overlay.events.mouse_over.constant(true);
+            is_hovered <+ model.overlay.events.mouse_out.constant(false);
+            let mouse_up = scene.mouse.frp.up.clone_ref();
+            let mouse_down = scene.mouse.frp.down.clone_ref();
+            let mouse_wheel = scene.mouse.frp.wheel.clone_ref();
+            let mouse_position = scene.mouse.frp.position.clone_ref();
+            caught_mouse <- any_(mouse_up,mouse_down,mouse_wheel,mouse_position);
+            pass_to_dom <- caught_mouse.gate(&is_hovered);
```

Comment:
I don't really like this desing. The thing is that:
1. The project owns navigator and documentation.
2. Then the documentation owns navigator and controls it (navigator is something inherently connected to the project).

So we have a logical circular dependency here. In reality, there is no dependency - project owns documentation AND navigator. So instead of this design, I think that documentation should send FRP output telling if its hovered or not. Then project, which owns documentation, should enable or disable navigator based on these events.

Reply:
Refactored as proposed